frontend/src/lib/k8s/clusterRole.ts

```diff
@@ -3,6 +3,11 @@ import { makeKubeObject } from './cluster';
 import { KubeRole } from './role';
 
 class ClusterRole extends makeKubeObject<KubeRole>('role') {
+  static kind = 'ClusterRole';
+  static apiName = 'clusterroles';
+  static apiVersion = 'rbac.authorization.k8s.io/v1';
+  static isNamespaced = false;
+
   static apiEndpoint = apiFactory('rbac.authorization.k8s.io', 'v1', 'clusterroles');
```

Comment:
This means we are repeating the apiEndpoint's args with the apiVersion and isNamespaced vars.
Would it work and be backward compat if we had apiEndpoint be a getter by default? So we could return apiFactory/WithNamespace using the apiVersion and isNamespaced vars.